zimbra/src/main/java/fr/openent/zimbra/service/impl/MessageService.java

```diff
@@ -336,10 +356,97 @@ private void transformMessageZimbraToFront(JsonObject msgZimbra, JsonObject msgF
      * @param msgFront   Front JsonObject
      * @param multiparts Array of multipart structure
      */
-    private void processMessageMultipart(JsonObject msgFront, JsonArray multiparts) {
+    private Future<Void> processMessageMultipart(JsonObject msgFront, JsonArray multiparts, UserInfos user) {
+        Promise<Void> promise = Promise.promise();
         Multipart mparts = new Multipart(msgFront.getString(FrontConstants.MESSAGE_ID), multiparts);
+
         msgFront.put(MESSAGE_BODY, mparts.getBody());
-        msgFront.put(MESSAGE_ATTACHMENTS, mparts.getAttachmentsJson());
+        replaceCidToUrl(mparts, msgFront, user).onComplete(ar -> {
+            if (ar.succeeded()) {
+                JsonObject updatedMsgFront = ar.result();
+                updatedMsgFront.put(MESSAGE_ATTACHMENTS, mparts.getAttachmentsJson());
+                msgFront.mergeIn(updatedMsgFront);
+                promise.complete();
+            } else {
+                // handle error
+                Throwable error = ar.cause();
+                log.error("Zimbra@processMessageMultipart : error processing multiparts: " + error.getMessage());
+                promise.fail(error);
```

Comment:
ne remonte pas la stacktrace, remonte ton message plutôt:
```suggestion
                String message = String.format("[Zimbra@%s::processMessageMultipart ] Fail to store " +
                                            "permissions preferences to session. %s", this.getClass().getSimpleName()));
                log.error(String.format("%s %s", message, error.getMessage());
                promise.fail(message);
```